pkg/middleware/requestmeta/request_metadata.go

```diff
@@ -57,8 +70,30 @@ func SetOwner(team string) web.Handler {
 	}
 }
 
-func defaultRequestMetadata() *RequestMetaData {
-	return &RequestMetaData{
-		Team: TeamCore,
+// WithDownstreamStatusSource sets the StatusSource field of the [RequestMetaData] for the
+// context to [StatusSourceDownstream].
+func WithDownstreamStatusSource(ctx context.Context) {
+	v := GetRequestMetaData(ctx)
+	v.StatusSource = StatusSourceDownstream
+}
+
+// WithStatusSource sets the StatusSource field of the [RequestMetaData] for the
+// context based on the provided statusCode.
+// If statusCode >= 500 then [StatusSourceDownstream].
+// If statusCode < 500 then [StatusSourceServer].
+func WithStatusSource(ctx context.Context, statusCode int) {
+	v := GetRequestMetaData(ctx)
```

Comment:
Any worry for these functions where we might override a `StatusSource` that has already been set?

Reply:
Not really for now. The default would be `server`. Similar concerns as raised in https://github.com/grafana/grafana/pull/74114#discussion_r1317400671 and concluded that we might need to follow up on this eventually.